clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredLoginCallbackHandler.java

```diff
@@ -279,7 +287,9 @@ private String claimOrHeaderJsonArrayText(String claimName, String escapedClaimV
     }
 
     private String escape(String jsonStringValue) {
-        return jsonStringValue.replace("\"", "\\\"").replace("\\", "\\\\");
+        return DOUBLE_BSLASH_REPLACE.matcher(
```

Comment:
This nested invocation is much harder to read than the original version. I believe storing the result of the first replace in a local variable would make it clearer.